File: 2019/ruby/test/aoc2019/day01_test.rb
require 'minitest/autorun'
require_relative '../../lib/aoc2019/day01.rb'

module Aoc2019::Test
  class Day01 < MiniTest::Test
    def part_one_line
      assert_equal 2, Aoc2019::Day01.line(12)
      assert_equal 2, Aoc2019::Day01.line(14)
      assert_equal 654, Aoc2019::Day01.line(1969)
      assert_equal 33583, Aoc2019::Day01.line(100756)
    end

    def test_part_one
      assert_equal 34241, Aoc2019::Day01.part_one("12\n14\n1969\n100756\n")
    end

    def test_part_two_line
      assert_equal 2, Aoc2019::Day01.line2(12)
      assert_equal 2, Aoc2019::Day01.line2(14)
      assert_equal 966, Aoc2019::Day01.line2(1969)
      assert_equal 50346, Aoc2019::Day01.line2(100756)
    end

    def test_part_two
      assert_equal 51316, Aoc2019::Day01.part_two("12\n14\n1969\n100756\n")
    end
  end
end

File: 2019/elixir/test/Aoc2019/Day01_test.exs
defmodule Aoc2019.Day01Test do
  use ExUnit.Case

  test "Part One Mass" do
    assert Aoc2019.Day01.mass(12) == 2
    assert Aoc2019.Day01.mass(14) == 2
    assert Aoc2019.Day01.mass(1969) == 654
    assert Aoc2019.Day01.mass(100_756) == 33583
  end

  test "Part One Complete" do
    assert Aoc2019.Day01.part_one("12\n14\n1969\n100756\n") == 34241
  end

  test "Culminative mass" do
    assert Aoc2019.Day01.culm_mass(12) == 2
    assert Aoc2019.Day01.culm_mass(14) == 2
    assert Aoc2019.Day01.culm_mass(1969) == 966
    assert Aoc2019.Day01.culm_mass(100_756) == 50346
  end

  test "Part Two" do
    assert Aoc2019.Day01.part_two("12\n14\n1969\n100756\n") == 51316
  end
end
